### PR TITLE
Hadron Id Submodule: initial release & deprecate Halo Id submodule

### PR DESCRIPTION
**Pull request by anthonylauzon**
## Type of change
- [X] Refactoring (no functional changes, no api changes)

## Description of change
This change fixes the naming of Audigent's Halo submodules to fit the new Hadron branding.

## Contact
prebid@audigent.com

## Docs Update Link
https://github.com/prebid/prebid.github.io/pull/3509

## Other information
https://github.com/prebid/Prebid.js/pull/7060
https://github.com/prebid/Prebid.js/pull/5777
https://github.com/prebid/Prebid.js/pull/6742

### CODE REVIEW

**Comment by anthonylauzon** [comment]
@ChrisHuie simple rebrand that requires a minor update for a couple of other adapters

**Comment by anthonylauzon** [comment]
@bretg code side review needed here - thanks

**Review by Fawke — CHANGES_REQUESTED** [review]
Hi @anthonylauzon,

Do we need a docs update here?

Also, can you please revert the changes made to the package-lock.json file?

**Comment by anthonylauzon** [comment]
Docs are approved here: https://github.com/prebid/prebid.github.io/pull/3509

package lock reverted

**Comment by patmmccann** [comment]
This appears to be a breaking change

**Comment by bretg** [comment]
Patrick is right - this would force every publisher currently using halo to change their in-page config. That's not ideal.

We don't currently support aliases for modules, but perhaps the new hadron module could deal with old "halo" config until 7.0?

**Comment by anthonylauzon** [comment]
I'm just as happy to duplicate the module until 7.0 so that there is a clear, seamless migration path.

**Comment by anthonylauzon** [comment]
That said, the change is not breaking - it is the same as deprecating any other module.  Publishers who are currently building the halo modules will need to rename the module configs in their build configurations accordingly.

**Comment by anthonylauzon** [comment]
I've temporarily re-added the Halo modules for continuity before 7.0, at which point they can be removed.

**Comment by anthonylauzon** [comment]
@patmmccann @Fawke @bretg this is ready for merge

**Review by Fawke — APPROVED** [review]
Looks good.
@patmmccann Do you mind having a second look? Thanks.

**Comment by Fawke** [comment]
> I've temporarily re-added the Halo modules for continuity before 7.0, at which point they can be removed.

@anthonylauzon I just realised that you didn't add the test files back for halo. Can you please include them as well so as to maintain a healthy test coverage for the overall project?

**Comment by patmmccann** [comment]
no longer looks breaking to me, but yes we shouldnt drop the test files

File: modules/beachfrontBidAdapter.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const SUPPORTED_USER_IDS = [
   { key: 'tdid', source: 'adserver.org', rtiPartner: 'TDID', queryParam: 'tdid' },
   { key: 'idl_env', source: 'liveramp.com', rtiPartner: 'idl', queryParam: 'idl' },
   { key: 'uid2.id', source: 'uidapi.com', rtiPartner: 'UID2', queryParam: 'uid2' },
-  { key: 'haloId', source: 'audigent.com', atype: 1, queryParam: 'haloid' }
+  { key: 'hadronId', source: 'audigent.com', atype: 1, queryParam: 'hadronid' }
```
Comment (patmmccann) [inline modules/beachfrontBidAdapter.js:25]: this is breaking, nevermind

**Review by patmmccann — CHANGES_REQUESTED** [review]
all the ssp adapters you removed halo support from are a blocker here

File: modules/ixBidAdapter.js [hunk 2]
```diff
@@ -57,14 +57,14 @@ const SOURCE_RTI_MAPPING = {
...
-  'audigent.com': '', // Halo ID from Audigent, haloId
+  'audigent.com': '', // Hadron ID from Audigent, hadronId
   'pubcid.org': '' // SharedID, pubcid
 };
 const PROVIDERS = [
   'britepoolid',
   'id5id',
   'lipbid',
-  'haloId',
+  'hadronId',
```
Comment (patmmccann) [inline modules/ixBidAdapter.js:67]: definitely breaking....

File: modules/openxBidAdapter.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ export const USER_ID_CODE_TO_QUERY_ARG = {
   britepoolid: 'britepoolid', // BritePool ID
   criteoId: 'criteoid', // CriteoID
   fabrickId: 'nuestarid', // Fabrick ID by Nuestar
-  haloId: 'audigentid', // Halo ID from Audigent
+  hadronId: 'audigentid', // Hadron ID from Audigent
```
Comment (patmmccann) [inline modules/openxBidAdapter.js:21]: breaking

**Comment by anthonylauzon** [comment]
test file re-added, reverted SSP changes and will handle bid adapter updates via separate PR(s)

**Comment by patmmccann** [comment]
you appear to still be modifying the index adapter

**Review by patmmccann — CHANGES_REQUESTED** [review]
it doesnt seem you reverted all breaking adapter changes

**Comment by anthonylauzon** [comment]
the only reference to hadron in the ix bid adapter appeared to be in a comment.  inconsistent, so changed and reverted that and another reference.  should be good to merge now.

**Comment by patmmccann** [comment]
I don't see any more breaking changes but please note @Fawke requested you restore file test/spec/modules/haloIdSystem_spec.js

**Review by patmmccann — APPROVED** [review]
LGTM

**Comment by anthonylauzon** [comment]
re-added

**Pull request merged**